src/cache/cache.rs

```diff
@@ -204,8 +204,20 @@ pub trait Storage: Send + Sync {
 /// Get a suitable `Storage` implementation from the environment.
 pub fn storage_from_environment() -> Box<Storage> {
     if let Ok(bucket) = env::var("SCCACHE_BUCKET") {
-        trace!("Trying S3Cache({})", bucket);
-        match S3Cache::new(&bucket) {
+        let region = match env::var("SCCACHE_REGION") {
+            Ok(region) => if region != "us-east-1".to_string() {
```

Comment:
Minor, but you can make this slightly verbose by moving the `if` into a match guard, like:
```rust
let region = match env::var("SCCACHE_REGION") {
    Ok(ref region) if region != "us-east-1" => format!("-{}", region),
    _ => String::new(),
};
```

also `String::from("")` and `String::new()` are functionally equivalent, I don't know if there's a preference there in general Rust usage, but the latter is shorter, at least.

I still think it might be a little nicer to move this down into the match arm where `SCCACHE_ENDPOINT` is unset, since you're not using the result if a custom endpoint is set anyway..